packages/apollo-gateway/src/executeQueryPlan.ts

```diff
@@ -388,7 +389,7 @@ async function executeFetch<TContext>(
  */
 function executeSelectionSet(
   source: Record<string, any> | null,
-  selectionSet: SelectionSetNode,
+  selections: SelectionNode[],
```

Comment:
This is pretty much the only major change that was needed to make this all work. This function used to just reference `selectionSet.selections`. We instead now just pass in the `selections` directly. All the changes to this function are really just to accommodate this change